outlines/fsm/fsm.py

```diff
@@ -37,8 +41,23 @@ class StopAtTokenFSM(FSM):
 
     def __init__(self, tokenizer: "Tokenizer", stop_token_id: int):
         self.stop_token_id = stop_token_id
-        self.vocabulary = tokenizer.vocabulary.values()
-        self.final_states = {1}
+        self.tokenizer = tokenizer
+        self.vocabulary = tokenizer.vocabulary
+        self.final_states = {2}
+        self.valid_alignment_tokens: List[int] = []
+
+    def align_prompt_tokens(self, prompt: str) -> str:
+        """Remove the last token from the prompt and set the value of self.valid_alignment_tokens"""
+        token_ids, _ = self.tokenizer.encode(prompt)
+        last_token_id = int(token_ids[0][-1])
+        last_token_text = self.tokenizer.decode([last_token_id])[0]
```

Comment:
I am not a huge fan of encoding/decoding here. I think we should return `token_ids` and use those to initialize the generation later on.